hr_employee_calendar_planning/models/resource_calendar.py

```diff
@@ -23,6 +23,21 @@ def _check_active(self):
                     % (item.name, total_items)
                 )
 
+    @api.constrains("company_id")
+    def _check_company_id(self):
+        for item in self.filtered("company_id"):
+            total_items = self.env["hr.employee.calendar"].search_count(
+                [
+                    ("calendar_id.company_id", "=", item.company_id.id),
+                    ("employee_id.company_id", "!=", item.company_id.id),
+                ]
+            )
+            if total_items:
+                raise ValidationError(
+                    _("%s is used in %s employee(s) related to other company.")
```

Comment:
I think this would be a bit clearer:

```suggestion
                    _("%s is used in %s employee(s) related to another company.")
```